pkg/provider/terraform/instance/plugin.go

```diff
@@ -50,7 +53,7 @@ type plugin struct {
 }
 
 // NewTerraformInstancePlugin returns an instance plugin backed by disk files.
-func NewTerraformInstancePlugin(dir string, pollInterval time.Duration, standalone bool) instance.Plugin {
+func NewTerraformInstancePlugin(dir string, pollInterval time.Duration, standalone bool, bootstrapGrpSpecStr, bootstrapInstID string) instance.Plugin {
```

Comment:
I have some ideas on organizing this.   These two params are probably better grouped together -- let's have a struct called `bootstrap` that has the fields `InstanceID` and `GroupSpecURL`...   Then have the struct be a parameter in the function.  All the logic here can then be moved into a function that takes a `bootstrap` struct as arg or have a method (where a `*bootstrap` is the receiver).  This would make the "constructor" function a little easier to follow and makes it easy to isolate all the bootstrap functionality for tests.   We also wont end up one day with a function that has 100 parameters ;)

If the `bootstrap` struct is optional, I'd make the function arg be a pointer `*bootstrap` and then check for nil value.   

By the way, can you add in the comments for this PR an example of the group spec you've used?   I noticed only a `TFormat` (from the instance plugin's properties) is used even though it required the entire group spec...  I personally prefer to have the code in the import / bootstrap be very specific about what it needs and then push all the extra parsing code to the code outside of this function.  In other words, if the logic here requires only a TFormat object, then have the caller figure out how to feed it, rather than giving it some big encompassing struct and have this doing all the work of parsing it and extracting just a little piece.   I think making explicit requirements on the data the function needs to do the work would make it much easier to follow what's going on.